tests/test_schema.py

```diff
@@ -0,0 +1,264 @@
+        WidgetToAddIndex.meta_.post_create()
+        WidgetToAddIndex.meta_.validate_model()
+        WidgetToAddIndex.meta_.post_validate()
+
+        changed = self.engine.update_schema()
+        self.assertListEqual(changed, [WidgetToAddIndex.meta_.ddb_tablename(self.engine.namespace)])
+
+        table = self.engine.dynamo.describe_table(WidgetToAddIndex.meta_.ddb_tablename(self.engine.namespace))
+        self.assertEqual(len(table.global_indexes), 2)
+
+        one = WidgetToAddIndex("one-1", "one-2", 1)
+        two = WidgetToAddIndex("one-2", "test", 2)
+
+        self.engine.save(one)
+        self.engine.save(two)
+
+        result = self.engine.query(WidgetToAddIndex).index('gindex-2').filter(WidgetToAddIndex.string2 == 'test').all()
+
+        self.assertEqual(len(result), 1)
+        self.assertNotEqual(result[0], one)
+        self.assertEqual(result[0], two)
+
+    def test_add_index_test_mode(self):
+        """Simulates adding a global secondary index at a later time. However, with the test attribute selected, no
+        change is actually made."""
+        table = self.engine.dynamo.describe_table(WidgetToAddIndex.meta_.ddb_tablename(self.engine.namespace))
+        self.assertEqual(len(table.global_indexes), 1)
+
+        # Simulating adding an index later on.
+        WidgetToAddIndex.meta_.global_indexes.append(GlobalIndex('gindex-2', 'string2', 'string'))
+        WidgetToAddIndex.meta_.post_create()
+        WidgetToAddIndex.meta_.validate_model()
+        WidgetToAddIndex.meta_.post_validate()
+
+        changed = self.engine.update_schema(test=True)
+        self.assertListEqual(changed, [WidgetToAddIndex.meta_.ddb_tablename(self.engine.namespace)])
+
+        table = self.engine.dynamo.describe_table(WidgetToAddIndex.meta_.ddb_tablename(self.engine.namespace))
+        self.assertEqual(len(table.global_indexes), 1)
+
+        try:
+            self.engine.query(WidgetToAddIndex).index('gindex-2').filter(WidgetToAddIndex.string2 == 'test').all()
+        except DynamoDBError:
+            pass
+        else:
+            assert False, "An error was expected"
+
+    def test_no_changes(self):
+        """ Tests that no index changes are performed """
+        table = self.engine.dynamo.describe_table(WidgetToAddIndex.meta_.ddb_tablename(self.engine.namespace))
+        self.assertEqual(len(table.global_indexes), 1)
+        changed = self.engine.update_schema(test=True)
+        self.assertListEqual(changed, [])
+        table = self.engine.dynamo.describe_table(WidgetToAddIndex.meta_.ddb_tablename(self.engine.namespace))
+        self.assertEqual(len(table.global_indexes), 1)
+
+    def test_alter_throughput_from_defaults(self):
+        """ Updates index throughput values by comparing the current values to those from the actual database.
+
+        This might not be a great idea. create_schema doesn't seem to perform any updates with read/write
+        throughput capacity changes. Not really sure if this should.
```

Comment:
I think this makes sense. `create_schema` will only perform actions if the table does not yet exist, but `update_schema` will sync the indexes and throughput. Reasonable design choice.